Clarifications regarding certificate chains. (#30)
tireddy2/pqc_uta: draft-ietf-uta-pqc-app.md

For a certification path to provide post-quantum authentication, each validation-critical signature from the end-entity certificate up to the configured trust anchor needs to be protected by a PQC or PQ/T hybrid signature scheme. A PQC end-entity certificate issued under a purely traditional CA hierarchy does not by itself provide post-quantum authentication, because an attacker equipped with a CRQC could potentially forge signatures made with traditional CA keys.

During migration, deployments may use separate PQC trust anchors, composite trust anchors, or parallel traditional and PQC certification paths. Clients need local policy to determine which trust anchors are acceptable for a given application and whether a traditional-only path is still permitted as fallback. Such fallback policies should be explicit, because accepting a traditional-only certification path can silently downgrade the authentication security property.

> Trust-anchor provisioning and update mechanisms are therefore a central part of PQC authentication migration. Long-lived or difficult-to-update systems, such as embedded devices and industrial deployments, may need to install PQC or hybrid trust anchors before PQC end-entity certificates are widely deployed.

Reviewer: ```suggestion
Trust-anchor provisioning and update mechanisms are therefore a central part of PQC authentication migration. Long-lived or difficult-to-update systems, such as embedded devices and industrial deployments, may need PQC or hybrid trust anchors provisioned early, potentially at manufacture, because later field updates may be infeasible. The trust anchor must be in place before the PQC end-entity certificates that chain to it come into use.
```

* Abridged TLS Certificate ({{?I-D.ietf-tls-cert-abridge}}): This approach minimizes the size of the certificate chain by omitting intermediate certificates that are already known to the client. Instead, the server provides a compact representation of the certificate chain, and the client reconstructs the omitted certificates using a well-known common CA database. This mechanism significantly reduces bandwidth requirements while preserving compatibility with existing certificate validation processes. Additionally, it explores potential methods to compress the end-entity certificate itself, though this aspect remains under discussion within the TLS Working Group.

> * Trust Anchor Identifiers ({{?I-D.ietf-tls-trust-anchor-ids}}): This extension allows a client to signal a compact list of locally configured trust anchors using unique trust anchor identifiers rather than Distinguished Names. This reduces the size of the "certificate_authorities" extension and helps the server select an appropriate certificate chain, especially when multiple hierarchies are used (e.g., separate traditional, composite, and PQC roots). Trust Anchor Identifiers are an optimization and chain-selection mechanism; they do not establish trust anchors, change the client's trust-store policy, or by themselves make a certification path quantum-ready. The client still has to validate the selected path against locally configured trust anchors and local policy for traditional, composite, or PQC authentication.

Reviewer: ```suggestion
* Trust Anchor Identifiers ({{?I-D.ietf-tls-trust-anchor-ids}}): This extension allows a client to signal a compact list of locally configured trust anchors using unique trust anchor identifiers rather than Distinguished Names. This reduces the size of the "certificate_authorities" extension and helps the server select an appropriate certificate chain, especially when multiple hierarchies are used (e.g., separate traditional, composite, and PQC roots). Trust Anchor Identifiers are an optimization and chain-selection mechanism; they do not establish trust anchors, change the client's trust-store policy, or by themselves make a certification path quantum-ready. The client still has to validate the path presented by the server against its locally configured trust anchors and local policy.
```

> * Trust Anchor Identifiers ({{?I-D.ietf-tls-trust-anchor-ids}}): This extension allows a client to signal a compact list of locally configured trust anchors using unique trust anchor identifiers rather than Distinguished Names. This reduces the size of the "certificate_authorities" extension and helps the server select an appropriate certificate chain, especially when multiple hierarchies are used (e.g., separate traditional, composite, and PQC roots). Trust Anchor Identifiers are an optimization and chain-selection mechanism; they do not establish trust anchors, change the client's trust-store policy, or by themselves make a certification path quantum-ready. The client still has to validate the selected path against locally configured trust anchors and local policy for traditional, composite, or PQC authentication.

Reviewer: Removed the trailing sentence on local policy deciding traditional vs. PQC acceptance: a client only advertises trust anchors it actually trusts, so the server must select a chain under a trust anchor already acceptable to the client.
